src/verify.rs

```diff
@@ -350,8 +382,8 @@ fn execute_step(state: &mut VerifyState, index: usize) -> Result<()> {
     // else we'll ignore the $e), and that logical file order is reflected in
     // the stack order of the hypotheses, we can do this in one pass
     for (ix, hyp) in fref.hypotheses.iter().enumerate() {
-        let slot = &state.stack[sbase + ix];
-
+        let (ref data, ref slot) = state.stack[sbase + ix];
+        datavec.push(data.clone());
```

Comment:
Not completely, no. But the hit is minimal (it should just be an integer for the length which is getting incremented).